server/src/resources/js/checkout.ts

```diff
@@ -0,0 +1,233 @@
+                email: "${values['email']}",
+                telephone: "${values['telephone']}",
+                mobile: "${values['mobile']}",
+                billing_address: {
+                    line_1: "${values['billing_address_line_1']}",
+                    line_2: "${values['billing_address_line_2']}",
+                    state: "${values['billing_address_state']}",
+                    postcode: "${values['billing_address_postcode']}",
+                    country: "${values['billing_address_country']}"
+                },
+                shipping_address: {
+                    line_1: "${values['shipping_address_line_1']}",
+                    line_2: "${values['shipping_address_line_2']}",
+                    state: "${values['shipping_address_state']}",
+                    postcode: "${values['shipping_address_postcode']}",
+                    country: "${values['shipping_address_country']}"
+                },
+                use_shipping_for_billing: ${values['use_shipping_for_billing']},
+                payment_method: "${values['payment_method']}",
+                agree_to_terms: ${values['agree_to_terms']},
+                notes: "${values['notes']}"
+            }){
+                event_data
+                order {
+                    id
+                }
+            }
+        }`;
+
+        return this.graphqlService.sendQuery(query, userToken, this.url);
+    }
+
+    public getBasketTotal() {
+        var userToken = localStorage.getItem('userToken') || '';
+        var self = this;
+
+        let query = `{
+            me {
+                cart {
+                    cart_total
+                }
+            }
+        }`;
+
+        window.checkoutCartTotal = () => {
+            return {
+                total: [],
+                fetch() {
+                    self.graphqlService.sendQuery(query, userToken, self.url).then((re) => {
+                        let total = re.data.me.cart.cart_total;
+                        this.total = total;
+                    });
+                },
+                formatPriceAndCurrency(cents: number) {
+                    let currency = localStorage.getItem('currency');
+
+                    // get the default currency from the shopConfig
+                    if (!currency) {
+                        currency = localStorage.getItem('default_currency');
+                    }
+
+                    return cents / 100 + ' ' + currency;
+                },
+            };
+        };
+    }
+
+    public displayLineItems() {
+        this.cart.endpointUrl(this.url);
+        let self = this;
+        let userToken = localStorage.getItem('userToken') || '';
+        let query = `{
+            me {
+                cart {
+                    items {
+                        product_id
+                        quantity
+                    }
+                }
+            }
+        }`;
+
+        window.lineItems = () => {
```

Comment:
Public interface method can also be done here, for consistency.

Reply:
completed